docs/proposals/scale_group_upgrade.md

```diff
@@ -0,0 +1,140 @@
+upgrade path for scale groups.
+
+## Rationale
+We currently provide an upgrade path for in-place upgrades.  We also need to provide a path
+for golden images when placed in scale groups.
+
+## Design
+The core principle here is to take advantage of the immutable infrastructure that is provided
+when working with golden images.
+
+The current playbooks in openshift ansible create autoscaling groups for infra/compute and 
+allow for custom groups as well.  This code lives in `role/openshift_aws`.
+
+Assume we are upgrading the compute node group:
+compute 123
+
+The upgrade process would look like this:
+* Create a new scale group `compute 456`
+* Scale up the new scale group to match the first groups capacity
+* Allow nodes to join the cluster by approving node CSRs
+* Set the nodes from `compute 123` as unschedulable
+* Drain the `compute 123` pods from the nodes
+* Determine if we want to role back to `compute 123` or continue on with `compute 456`.
+* Remove scale group `compute 123` or `compute 456`
+
+## Benefits
+
+There are many benefits to this approach.
+
+* Treat the scale groups as immutable infrastructure
+* Roll back strategy is as simple as scaling the old scale group back to capacity
+* Managing hosts can be done by scale group or by scale group tags
+
+## Downside
+
+* Managing the compute name or version
+* Risk of removing scale groups
+* Process surrounding acceptance of nodes (not specific to this strategy)
+
+## User Story
+As an admin of OpenShift-Ansible,
+I want to upgrade my scale groups
+so that I can receive product updates, CVE remediations, and provide a path forward for customers.
+
+## Acceptance Criteria
+* Verify that a new scale group is created
+* Verify that the old scale group properly drained
+* Verify that the old scale group is removed
+* Verify that the old launch config is removed
+* Verify there is documentation about the process
+
+## References
+
+
+## Workflow
+
+Outline of a typical workflow for an upgrade.
+* An AMI is built with a specific version of openshift
+  This is done through openshift-ansible/playbooks/aws/openshift-cluster/build_ami.yml playbook.
+  Specify the correct repository location and version via the following vars:
+  openshift_pkg_version=
+  openshift_additional_repositories=[]
+* The output of the build_ami should be an AMI that is placed in the AWS account.
+  The following upgrade playbooks will handle the AMI in two ways:
+  * If the openshift_aws_ami is specified then the scale groups will use the specific id.
+  * If multiple AMIs are desired then you can place them in the the openshift_aws_ami_map:
+  ```
+  openshift_aws_ami_map:
+    master: "{{ openshift_aws_ami }}"
+    infra: "{{ openshift_aws_ami }}"
+    compute: "{{ openshift_aws_ami }}"
+  ```
+  * If the openshift_aws_ami is unspecified, the AWS code will search for the latest
+    AMI with the following name and use it:
+  ```
+  openshift_aws_ami_name: openshift-gi
```

Comment:
This is kinda confusing.  we went from openshift_aws_ami_name variable to openshift-gi.  Is this somethign the user needs to put in the inventory? 

I think cleaning up the wording would help me understand this better.

So far there are vars:

openshift_aws_ami
openshift_aws_ami_map
openshift_aws_ami_name

Reply:
`openshift_aws_ami_name` is the prefix name of the AMI that was built.  It is the default.  If you' prefer to have something else then you can specify that.

Hopefully this will clarify as well as the previous posts:
> openshift_aws_ami
This variable is the variable for the AMI id.  This can be specified or found dynamically.  If it is defined then it will be used for all node groups.
> openshift_aws_ami_map
This ami map is a dictionary of the node group types.  If you do nothing in your inventory it defaults to the value of `openshift_aws_ami`.  This means you either specify it or as mentioned above we attempt to find the latest version using the `openshift_aws_ami_name`.
> openshift_aws_ami_name
When an AMI is built it uses this prefix to create an AMI with this name.  When you are creating a launch config you will need to specify an AMI id.  You can specify it through the previous two methods or you can have it find the latest image by not specifying the `openshift_aws_ami`.  This will use the `openshift_aws_ami_name` to query AWS to find the latest AMI to use.